packages/nimble-components/src/combobox/index.ts

```diff
@@ -148,78 +348,352 @@ export class Combobox
-    public override focusoutHandler(e: FocusEvent): boolean | void {
-        const returnValue = super.focusoutHandler(e);
+    /**
+     * @internal
+     */
+    public keyupHandler(e: KeyboardEvent): boolean {
+        const key = e.key;
+
+        switch (key) {
+            case 'ArrowLeft':
+            case 'ArrowRight':
+            case 'Backspace':
+            case 'Delete':
+            case 'Home':
+            case 'End': {
+                this.filter = this.control.value;
+                this.selectedIndex = -1;
+                this.filterOptions();
+                break;
+            }
+            default: {
+                break;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @internal
+     */
+    public override focusoutHandler(e: FocusEvent): boolean {
+        this.syncValue();
+
+        if (this.open) {
+            const focusTarget = e.relatedTarget as HTMLElement;
+            if (this.isSameNode(focusTarget)) {
+                this.focus();
+            }
+        }
+
         this.open = false;
         this.emitChangeIfValueUpdated();
-        return returnValue;
+        return true;
+    }
+
+    /**
+     * Reset the element to its first selectable option when its parent form is reset.
+     *
+     * @internal
+     */
+    public override formResetCallback(): void {
+        super.formResetCallback();
+        this.setDefaultSelectedOption();
+        this.updateValue();
+    }
+
+    /** {@inheritDoc (FormAssociated:interface).validate} */
+    public override validate(): void {
+        super.validate(this.control);
+    }
+
+    /**
+     * Set the default selected options at initialization or reset.
+     *
+     * @internal
+     * @remarks
+     * Overrides `Listbox.setDefaultSelectedOption`
+     */
+    public override setDefaultSelectedOption(): void {
+        if (this.$fastController.isConnected && this.options) {
+            const selectedIndex = this.options.findIndex(
+                el => el.getAttribute('selected') !== null || el.selected
+            );
+
+            this.selectedIndex = selectedIndex;
+            if (!this.dirtyValue && this.firstSelectedOption) {
+                this.value = this.firstSelectedOption.text;
+            }
+            this.setSelectedOptions();
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public override selectedIndexChanged(
+        prev: number | undefined,
+        next: number
+    ): void {
+        if (this.$fastController.isConnected) {
+            const pinnedSelectedIndex = limit(
+                -1,
+                this.options.length - 1,
+                next
+            );
+
+            // we only want to call the super method when the selectedIndex is in range
+            if (pinnedSelectedIndex !== this.selectedIndex) {
+                this.selectedIndex = pinnedSelectedIndex;
+                return;
+            }
```

Comment:
May certainly be missing it but is this new logic (don't think I see this bit in the original change)? Do we have test coverage for it?

Reply:
This logic is a slight update to the [original](https://github.com/microsoft/fast/blob/0c27d027ff6e8616ad4fddc17f4432aa7f6cbad0/packages/web-components/fast-foundation/src/combobox/combobox.ts#L547). I had to make the change that I did to avoid the Typescript issue of trying to modify a method parameter.